terraform-tests/mssql_test.go

```diff
@@ -487,4 +489,48 @@ var _ = Describe("mssql", Label("mssql-terraform"), Ordered, func() {
 			})
 		})
 	})
+
+	Context("auto_minor_version_upgrade", func() {
+		When("is enabled and a not major version is selected", func() {
+			It("should complain about postcondition", func() {
+				session, _ := FailPlan(terraformProvisionDir, buildVars(defaultVars, requiredVars, map[string]any{
+					"auto_minor_version_upgrade": true,
+				}))
+
+				Expect(session.ExitCode()).NotTo(Equal(0))
+				msgs := string(session.Out.Contents())
+				Expect(msgs).To(ContainSubstring(`Error: Resource postcondition failed`))
+				Expect(msgs).To(ContainSubstring(`A Major engine version should be specified when auto_minor_version_upgrade is enabled. Expected engine version: 15.00 - got: 15.00.4236.7.v1`))
+			})
+		})
+
+		When("is disabled and a major version is selected", func() {
+			It("should not complain about postcondition", func() {
+				plan := ShowPlan(terraformProvisionDir, buildVars(defaultVars, requiredVars, map[string]any{
+					"auto_minor_version_upgrade": false,
+					"mssql_version":              "15.00",
+				}))
+
+				Expect(AfterValuesForType(plan, "aws_db_instance")).To(
+					MatchKeys(IgnoreExtras, Keys{
+						"auto_minor_version_upgrade": BeFalse(),
+						"engine_version":             Equal("15.00"),
+					}))
+			})
+		})
+
+		When("is disabled and a minor version is selected", func() {
+			It("should not complain about postcondition and create the instance", func() {
+				plan := ShowPlan(terraformProvisionDir, buildVars(defaultVars, requiredVars, map[string]any{
+					"auto_minor_version_upgrade": false,
+				}))
+
+				Expect(AfterValuesForType(plan, "aws_db_instance")).To(
+					MatchKeys(IgnoreExtras, Keys{
+						"auto_minor_version_upgrade": BeFalse(),
+						"engine_version":             Equal(requiredVars["mssql_version"]),
+					}))
+			})
+		})
```

Comment:
nitpick: although not needed, because it is defined in requiredVars, including an explicit value for mssql_version in the test might provide more clarity about the postcondition and its behaviour.

**Github highlights the whole test although I only changed the following:**
- added `mssql_version` to the list of parameters
- hardcoded `15.00.4236.7.v1` in the `MatchKeys` comparison for consistency with the previous bullet

```suggestion
		When("is disabled and a minor version is selected", func() {
			It("should not complain about postcondition and create the instance", func() {
				plan := ShowPlan(terraformProvisionDir, buildVars(defaultVars, requiredVars, map[string]any{
					"auto_minor_version_upgrade": false,
					"mssql_version":              "15.00.4236.7.v1",
				}))

				Expect(AfterValuesForType(plan, "aws_db_instance")).To(
					MatchKeys(IgnoreExtras, Keys{
						"auto_minor_version_upgrade": BeFalse(),
						"engine_version":             Equal("15.00.4236.7.v1"),
					}))
			})
		})
```

I have preferred to do a multiline suggestion instead of two suggestions to the affected lines to clarify that they come together. Both lines should be modified or none of them should.
But it has the less-than-ideal effect of showing up as a full rewrite in GitHub web interface 😔.

Reply:
Thank you very much! I love it!